File: finanze/application/ports/last_fetches_port.py
import abc
from uuid import UUID

from domain.fetch_record import FetchRecord


class LastFetchesPort(metaclass=abc.ABCMeta):
    async def get_by_entity_id(self, entity_id: UUID) -> list[FetchRecord]:
        raise NotImplementedError

    async def save(self, fetch_records: list[FetchRecord]):
        raise NotImplementedError

File: mobile/src/application/ports/core/lastFetchesPort.ts
import { Entity, Feature, FetchRecord } from "@/domain"

export interface LastFetchesPort {
  getByEntityId(entityId: string): Promise<FetchRecord[]>
  getGroupedByEntity(feature: Feature): Promise<Map<Entity, FetchRecord>>
  save(fetchRecords: FetchRecord[]): Promise<void>
}
